sys-boot/udk/udk-2018-r1.ebuild

```diff
@@ -0,0 +1,136 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 )
+PYTHON_REQ_USE="sqlite"
+
+inherit flag-o-matic multiprocessing python-single-r1 toolchain-funcs
+
+DESCRIPTION="Tianocore UEFI Development kit"
+HOMEPAGE="https://github.com/tianocore/tianocore.github.io/wiki/EDK-II"
+
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/tianocore/edk2.git"
+	KEYWORDS=""
+else
+	MY_V="${PN^^}${PV}"
+	SRC_URI="https://github.com/tianocore/edk2/archive/v${MY_V}.tar.gz -> ${P}.tar.gz
+		doc? ( https://github.com/tianocore/edk2/releases/download/v${MY_V}/${MY_V}.Documents.zip -> ${P}-docs.zip )"
+	S="${WORKDIR}/edk2-v${MY_V}"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="doc"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="strip"
+
+RDEPEND="${PYTHON_DEPS}"
+DEPEND="${RDEPEND}
+	>=dev-lang/nasm-2.14.02
+	>=sys-power/iasl-20160729
+	doc? ( app-arch/unzip )"
+
+DOCDIR="${WORKDIR}/Documents"
+
+# EFI pre-build libs
+QA_PREBUILT="
+	usr/lib*/${P}/BeagleBoardPkg/Debugger_scripts/rvi_dummy.axf
+	usr/lib*/${P}/ArmPkg/Library/GccLto/*.a
+"
+
+pkg_setup() {
+	if use x86; then
+		EFIARCH=IA32
+	elif use amd64; then
+		EFIARCH=X64
+	fi
+
+	# Select toolchain within predefined ones
+	if tc-is-gcc; then
+		TOOLCHAIN_TAG="GCC5"
+	elif tc-is-clang; then
+		TOOLCHAIN_TAG="CLANG38"
+	else
+		TOOLCHAIN_TAG="ELFGCC"
+	fi
+}
+
+src_unpack() {
+	default
+
+	local doc_name
+	local f
+	if use doc; then
+		pushd "${DOCDIR}" >/dev/null || die
+		rm -f *.chm || die
+		for f in *.zip; do
+			unpack "${DOCDIR}/${f}"
+			mv html "${f%.zip}" || die
+		done
+		rm -f *.zip || die
+		popd >/dev/null || die
+	fi
+}
+
+src_configure() {
+	sed -e "s:^\(BUILD_CFLAGS\s*=\).*$:\1 ${CFLAGS} -MD -fshort-wchar -fno-strict-aliasing -nostdlib -c -fPIC:" \
+		-i "BaseTools/Source/C/Makefiles/header.makefile" \
+		|| die "Failed to update makefile header"
+}
+
+src_compile() {
+	local make_flags=(
+		BUILD_CC="$(tc-getBUILD_CC)"
+		BUILD_CXX="$(tc-getBUILD_CXX)"
+		BUILD_AS="$(tc-getBUILD_AS)"
+		BUILD_AR="$(tc-getBUILD_AR)"
+		BUILD_LD="$(tc-getBUILD_LD)"
+	)
+	# Base tools does not like parallel make
+	emake "${make_flags[@]}" -j1 -C BaseTools
+
+	# Update template parameter files
+	sed -e "s:^\(ACTIVE_PLATFORM\s*=\).*$:\1 MdeModulePkg/MdeModulePkg.dsc:" \
+		-e "s:^\(TARGET\s*=\).*$:\1 RELEASE:" \
+		-e "s:^\(TARGET_ARCH\s*=\).*$:\1 ${EFIARCH}:" \
+		-e "s:^\(TOOL_CHAIN_TAG\s*=\).*$:\1 ${TOOLCHAIN_TAG}:" \
+		-e "s:^\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+		-i "BaseTools/Conf/target.template" || die "Failed to configure target file"
+
+	# Clean unneeded files
+	find . -name '*.bat' -exec rm -f {} \;
+	find ./BaseTools/Source/C -mindepth 1 -maxdepth 1 \! -name 'bin' -exec rm -rf {} \;
+
+	# Create workspace script file
+	mkdir "${WORKDIR}/script"
+	sed -e "s:{EDK_BASE}:${EPREFIX}/usr/$(get_libdir)/${P}:" \
+		"${FILESDIR}"/udk-workspace.template \
+		>"${WORKDIR}"/script/udk-workspace || die "Failed to build udk-workspace"
```

Comment:
We normally use `${T}` for this. There's really no need to create extra directories.